src/components/search/searchQueries.js

```diff
@@ -0,0 +1,141 @@
+/**
+ *
+ * @author sriram
+ *
+ *
+ */
+
+import { useQuery, useInfiniteQuery } from "react-query";
+import { searchData, searchDataInfinite } from "serviceFacades/filesystem";
+import { searchApps, searchAppsInfiniteQuery } from "serviceFacades/apps";
+import { getAnalyses, searchAnalysesInfinite } from "serviceFacades/analyses";
+
+/**
+ *
+ * Data search query
+ *
+ * @param {object} dataSearchKey - The query key to be used.
+ * @param {boolean} dataSearchQueryEnabled - Enable / diable query.
+ * @param {function} successCallback - Callback function to be used when the query succeeds.
+ *
+ * @returns {function}
+ */
+function useDataSearch(dataSearchKey, dataSearchQueryEnabled, successCallback) {
+    return useQuery({
+        queryKey: dataSearchKey,
+        queryFn: searchData,
+        config: {
+            enabled: dataSearchQueryEnabled,
+            onSuccess: (results) => successCallback(results),
+        },
+    });
+}
```

Comment:
Also, if the 2nd and 3rd param of each of these `use*Search` functions in this file were renamed to `enabled` and `onSuccess`, then the `config` could be simplified in each one:
```suggestion
function useDataSearch(dataSearchKey, enabled, onSuccess) {
    return useQuery({
        queryKey: dataSearchKey,
        queryFn: searchData,
        config: {
            enabled,
            onSuccess,
        },
    });
}
```

Reply:
Neat! ✨